spec/index.adoc

```diff
@@ -2165,22 +2165,27 @@ NOTE: Canisters not using these features can completely leave out the `streaming
 
 The Gateway needs to know the canister id of the canister to talk to, and obtains that information from the hostname as follows:
 
-1. Check that the hostname, taken from the `Host` field of the HTTP request, is of the form `<name>.raw.ic0.app` or `<name>.ic0.app`, or fail.
-
-2. If the `<name>` is in the following table, use the given canister ids:
+1. If the hostname is in the following table, use the given canister ids:
 +
 .Canister hostname resolution
-|============================================
-| Hostname     | Canister id
-| `identity`   | `rdmx6-jaaaa-aaaaa-aaadq-cai`
-| `nns`        | `qoctq-giaaa-aaaaa-aaaea-cai`
-| `dscvr`      | `h5aet-waaaa-aaaab-qaamq-cai`
-| `personhood` | `g3wsl-eqaaa-aaaan-aaaaa-cai`
-|============================================
+|=====================================================
+| Hostname             | Canister id
+| `identity.ic0.app`   | `rdmx6-jaaaa-aaaaa-aaadq-cai`
+| `nns.ic0.app`        | `qoctq-giaaa-aaaaa-aaaea-cai`
+| `dscvr.one`          | `h5aet-waaaa-aaaab-qaamq-cai`
+| `dscvr.ic0.app`      | `h5aet-waaaa-aaaab-qaamq-cai`
+| `personhood.ic0.app` | `g3wsl-eqaaa-aaaan-aaaaa-cai`
+|=====================================================
+
+2. Check whether the hostname is a _raw_ hostname (e.g., `<name>.raw.ic0.app`). If it is the case, fail and handle the request as a Web2 request, otherwise continue.
+
+3. Check whether the canister ID is embedded in the hostname by splitting the hostname and finding the first occurrence of a valid canister ID from the right. If there is a canister ID embedded in the hostname, use it.
```

Comment:
We also take the `topdomains` to determine the gateway to call.  

`rdmx6-jaaaa-aaaaa-aaadq-cai.ic0.app` will extract the principal and use `ic0.app` as the api gateway.